src/Quizzer.API.DAL/QuizzerDbContext.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Quizzer.API.DAL.Entities;
+
+namespace Quizzer.API.DAL {
+    public class QuizzerDbContext : DbContext {
+        public DbSet<QuizEntity> Quiezzes { get; set; } = null!;
+        public DbSet<CompletedQuizEntity> CompletedQuiezzes { get; set; } = null!;
+        public DbSet<QuestionEntity> Questions { get; set; } = null!;
+        public DbSet<AnswerEntity> Answers { get; set; } = null!;
+        public DbSet<SelectedAnswerEntity> SelectedAnswers { get; set; } = null!;
+        public DbSet<UserEntity> Users { get; set; } = null!;
+        public DbSet<UserAnswerEntity> UserAnswers { get; set; } = null!;
+
+        public QuizzerDbContext(DbContextOptions<QuizzerDbContext> options)
+            : base(options) {
+        }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder) {
+            base.OnModelCreating(modelBuilder);
+
+            modelBuilder.Entity<QuizEntity>()
+                .HasMany<CompletedQuizEntity>()
+                .WithOne(completedQuiz => completedQuiz.Quiz)
+                .OnDelete(DeleteBehavior.Cascade);
+
+            modelBuilder.Entity<QuizEntity>()
+                .HasMany<QuestionEntity>()
+                .WithOne(question => question.Quiz)
+                .OnDelete(DeleteBehavior.Cascade);
+            //
+            modelBuilder.Entity<UserEntity>()
+                .HasMany<CompletedQuizEntity>()
+                .WithOne(completedQuiz => completedQuiz.User)
+                .OnDelete(DeleteBehavior.Cascade);
+
+            modelBuilder.Entity<UserEntity>()
+                .HasMany<UserAnswerEntity>()
+                .WithOne(userAnswer => userAnswer.User)
+                .OnDelete(DeleteBehavior.Cascade);
+            //
+            modelBuilder.Entity<QuestionEntity>()
+                .HasMany<AnswerEntity>()
+                .WithOne(answer => answer.Question)
+                .OnDelete(DeleteBehavior.Cascade);
+
+            modelBuilder.Entity<QuestionEntity>()
+                .HasMany<UserAnswerEntity>()
+                .WithOne(userAnswer => userAnswer.Question)
+                .OnDelete(DeleteBehavior.Cascade);
+            //
+            modelBuilder.Entity<AnswerEntity>()
+                .HasMany<SelectedAnswerEntity>()
+                .WithOne(selectedAnswer => selectedAnswer.Answer)
+                .OnDelete(DeleteBehavior.Cascade);
+            //
+            modelBuilder.Entity<UserAnswerEntity>()
+                .HasMany<SelectedAnswerEntity>()
+                .WithOne(selectedAnswer => selectedAnswer.UserAnswer)
+                .OnDelete(DeleteBehavior.Cascade);
```

Comment:
I think there should also be some configuration indicating to EF that this is binding entity to three entities. 

Try to read about `UsingEntity` method

Information could be maybe found [here](https://learn.microsoft.com/en-us/ef/core/modeling/relationships/many-to-many)

Reply:
For clarity. I mean the `UserAnswer `entity, which is binding entity to three entities. When I looked at the ActiveTracker, we were adding `UsingEntity `configuration as well for binding entity...

Reply:
`UsingEntity` is used for "skip navigations" in this case `UserAnswerEntity` is join entity between `UserEntity `and `QuestionEntity` because of many to many relationship between `User `and `Question`.  A "skip navigation" would be for example `ICollection<QuestionEntity> AnsweredQuestions` property on `UserEntity`, then we would use `UsingEntity`. But we just use `ICollection<UserAnswerEntity>` on both, so `UsingEntity` should not be necessary.

Reply:
In case EF core doesn't catch it tests should reveal it.

Reply:
Ok I see. Then it is ok.